evap/results/views.py

```diff
@@ -92,8 +138,21 @@ def index(request):
     evaluations = [evaluation for evaluation in evaluations if evaluation.can_user_see_evaluation(request.user)]
 
     if request.user.is_reviewer:
-        additional_evaluations = Evaluation.objects.filter(course__semester__in=semesters, state__in=['in_evaluation', 'evaluated', 'reviewed'])
-        evaluations += get_evaluations_with_prefetched_data(additional_evaluations)
+        additional_evaluations = get_evaluations_with_prefetched_data(
+            Evaluation.objects.filter(
+                course__semester__in=semesters,
+                state__in=['in_evaluation', 'evaluated', 'reviewed']
+            )
+        )
+        for evaluation in additional_evaluations:
+            # the value is required in the template if the evaluation's course has multiple evaluations.
+            # as this evaluation isn't published, the course can have no average grade yet.
+            evaluation.course_avg_grade = None
+            evaluation.course_weight_sum = sum(evaluation.weight for evaluation in evaluation.course.evaluations.all())
+            evaluation.course.not_all_evaluations_are_published = True
```

Comment:
Does this not require a .save()? Same exists in some other places, too.

Reply:
the values are not stored in the db and are just passed on with this current instance